pkg/lib/_internal/packagemanager-abstract.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <https://www.gnu.org/licenses/>.
+ */
+
+export interface ProgressData {
+    // PackageManager waits on a lock or another operation to finish
+    waiting: boolean
+    // If not null, the operation can be cancelled
+    cancel?: (() => void) | null
+    // FIXME, mistake? should be percentage?
```

Comment:
Mistake where? That's just how PackageKit's callback names that, but of course the abstraction is free to rename it to just "percentage"?